WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift

```diff
@@ -13,106 +13,182 @@ struct CommentModerationView: View {
             Divider()
                 .foregroundStyle(Color.DS.Background.secondary)
             VStack(spacing: .DS.Padding.double) {
-                titleHStack
-                mainActionView
-                secondaryActionView
+                switch viewModel.state {
+                case .pending:
+                    Pending(viewModel: viewModel)
+                case .approved(let liked):
+                    Approved(viewModel: viewModel, liked: liked)
+                case .trash, .spam:
+                    TrashSpam(viewModel: viewModel)
+                }
```

Comment:
My understanding is that the previous implementation featured the moderation view as a single entity with three components:
- The title
- The main action view
- The secondary view

These subviews would change depending on the `viewModel.state`. I tried to fix the transition glitch without altering the overall structure too much, but it was challenging. Consequently, I've shifted to a new approach where:

- Each `viewModel.state` is now represented as a distinct view.
- Switching states involves simply transitioning between these views.